tools/azure-sdk-tools/devtools_testutils/storage_testcase.py

```diff
@@ -137,4 +141,44 @@ def _get_resource_group(self, **kwargs):
             raise AzureTestError(template.format(ResourceGroupPreparer.__name__))
 
 
+class BlobAccountPreparer(StorageAccountPreparer):
+    def __init__(
+        self,
+        **kwargs
+    ):
+        self.is_versioning_enabled = kwargs.pop("is_versioning_enabled", None)
+        self.is_last_access_time_enabled = kwargs.pop("is_last_access_time_enabled", None)
+        self.container_retention_days = kwargs.pop("container_retention_days", None)
+
+        super(BlobAccountPreparer, self).__init__(**kwargs)
+
+    def _create_account(self, resource_group_name, account_name):
+        storage_async_operation = self.client.storage_accounts.begin_create(
+            resource_group_name,
+            account_name,
+            {
+                "sku": {"name": self.sku},
+                "location": self.location,
+                "kind": self.kind,
+                "enable_https_traffic_only": True,
+            },
+        )
+
+        props = BlobServiceProperties()
+        if self.is_versioning_enabled is True:
+            props.is_versioning_enabled = True
+        if self.container_retention_days:
+            props.container_delete_retention_policy = DeleteRetentionPolicy(enabled=True,
+                                                                            days=self.container_retention_days)
+        if self.is_last_access_time_enabled:
+            props.last_access_time_tracking_policy = LastAccessTimeTrackingPolicy(enable=True)
+
+        if not all(prop is None for prop in props.as_dict().values()):
+            self.client.blob_services.set_service_properties(resource_group_name, account_name, props)
+
+        sleep(30)
```

Comment:
Is this sleep needed? If so, why?

Reply:
it looks like those service properties need some time to take effect, though the time varies, 30s is fine for now